File: 04.Navigation/MyNavigationDrawer/app/src/main/java/com/dicoding/mynavigationdrawer/PageFragment.java
package com.dicoding.mynavigationdrawer;


import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;


/**
 * A simple {@link Fragment} subclass.
 */
public class PageFragment extends Fragment {

    static final String TAG = PageFragment.class.getSimpleName();

    public static final String EXTRAS = "extras";

    public PageFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_page, container, false);

        TextView textView = view.findViewById(R.id.tv_fragment);

        /*
        Ambil data dari arguments yang dikirimkan oleh activity dengan mengecek terlebih dahulu apakah data tidak bernilai null
         */

        if (getArguments() != null) {
            String page = getArguments().getString(EXTRAS);
            textView.setText(page);

            Log.e(TAG, "onCreateView: halaman fragment " + page);
        }

        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        if (getArguments() != null && getActivity() != null) {
            String title = getArguments().getString(EXTRAS);

            ((AppCompatActivity) getActivity()).getSupportActionBar().setTitle(title);
        }
    }
}

File: Navigation/MyNavigationDrawer/kotlin/app/src/main/java/com/dicoding/picodiploma/mynavigationdrawer/PageFragment.kt
package com.dicoding.picodiploma.mynavigationdrawer

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import kotlinx.android.synthetic.main.fragment_page.*

/**
 * A simple [Fragment] subclass.
 */
class PageFragment : Fragment() {

    companion object {
        const val EXTRAS = "extras"
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_page, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        /*
        Ambil data dari arguments yang dikirimkan oleh activity
         */
        if (arguments != null) {
            val title = arguments?.getString(EXTRAS)

            // Update string textview
            tv_fragment.text = title

            val activity = activity
            if (activity != null) {
                val actionBar = (activity as AppCompatActivity).supportActionBar
                // Update title actionbar-nya
                if (actionBar != null) actionBar.title = title
            }
        }
    }
}